pkg/queue/health/probe.go

```diff
@@ -202,3 +219,69 @@ func isHTTPProbeReady(res *http.Response) bool {
 	// response status code between 200-399 indicates success
 	return res.StatusCode >= 200 && res.StatusCode < 400
 }
+
+// GRPCProbe checks that gRPC connection can be established to the address.
+func GRPCProbe(config GRPCProbeConfigOptions) error {
+	dialer := &net.Dialer{
+		Control: func(network, address string, c syscall.RawConn) error {
+			return c.Control(func(fd uintptr) {
+				unix.SetsockoptLinger(int(fd), syscall.SOL_SOCKET, syscall.SO_LINGER, &unix.Linger{Onoff: 1, Linger: 1})
+			})
+		},
+	}
+
+	v := version.Get()
```

Comment:
This doesn't work - this package is from the k8s code and assumes the binary is built with certain linker flags.

I'd look what we're doing with the HTTPProbe